### DISCUSSION TITLE
HTML output improvements

### DISCUSSION THREAD

**Discussion by QJKX** [linkchecker/linkchecker#803] [Ideas]
Semantically I think the output is "wrong" - the errors should be all be in the same table, not a new table for each error.

Actually displaying this would seem to be impractical because of column width. One approach is [using CSS to stack the columns if the screen is too narrow](https://stackoverflow.com/questions/55976255/table-column-is-too-wide/55976351#55976351)  ([example](http://scratchpad.io/messy-eyes-6557)) 

With everything in one table sorting can be added. For sorting there should probably be a timestamp column, otherwise it's not obvious how to get back to the original state short or reloading.

In the spirit of URLs each row should have an anchor enabling deep links to a specific error.

The Parent URL item could hyperlink using the [scroll to text fragment](https://caniuse.com/url-scroll-to-text-fragment) standard.

Thoughts?

Would people object to this being implemented with jinja2 templates?

**Comment by ajmalfaris11** [comment]
Your proposed approach sounds reasonable and aligns well with common practices for displaying tabular data with potential sorting and hyperlinking capabilities. Here are some thoughts and considerations:

1. **Single Table for Errors**: Consolidating all errors into a single table is a good idea for better organization and presentation of data, especially if there are multiple errors to display.

2. **Responsive Design with CSS**: Using CSS to make the table responsive, such as stacking columns on narrow screens, ensures a better user experience across different devices and screen sizes.

3. **Sorting and Timestamp Column**: Adding a timestamp column for sorting and enabling users to revert to the original state is a useful feature. Sorting options can enhance usability, especially if there is a large amount of data.

4. **Deep Links and Anchors**: Providing anchors for each row and deep linking to specific errors improves navigation and allows users to share or bookmark specific error instances easily.

5. **Hyperlinks for URLs**: Hyperlinking the Parent URL item using the scroll-to-text-fragment standard enhances usability and enables users to navigate directly to the relevant portion of the page.

6. **Implementation with Jinja2 Templates**: Implementing this approach with Jinja2 templates is a common and practical choice, especially if you're already using Jinja2 in your web application. Jinja2 templates provide flexibility and ease of integration with Python-based web frameworks like Flask or Django.

Overall, your proposed approach seems well-considered and user-friendly. Implementing it with Jinja2 templates should be feasible and may even simplify the development process due to Jinja2's templating capabilities and integration with Python web frameworks.

**Reply by QJKX** [reply]
🤖

**Comment by lukajincharadze** [comment]
Your suggestions for improving the presentation and functionality of the error display are thoughtful and would likely enhance the user experience. Let's address each point:

Single Table for Errors: Aggregating all errors into a single table would provide a more coherent and organized view for users to review and manage errors effectively.

Responsive Design: Implementing CSS to stack columns on narrow screens would improve readability and usability across different devices and screen sizes.

Sorting and Timestamps: Adding a timestamp column would facilitate sorting and provide users with the ability to revert to the original state if needed. Sorting functionality would enhance the usability of the error display.

Deep Links: Including anchors for each row would enable users to deep link to specific errors, enhancing navigation and usability, especially in large error lists.

Hyperlinking Parent URLs: Linking the Parent URL column using the scroll to text fragment standard would improve usability and enable users to quickly navigate to the relevant context for each error.

Implementing these enhancements using Jinja2 templates should be feasible and would likely be well-received by users. Jinja2 templates provide a convenient way to generate dynamic HTML content and integrate server-side logic, making them suitable for rendering error displays with the desired functionality.

Overall, these improvements align with best practices for usability and accessibility, and they should contribute positively to the user experience of the error display.

**Reply by QJKX** [reply]
🤖

**Reply by lukajincharadze** [reply]
??

**Comment by QJKX** [comment]
No constructive discussion, just LLM word vomit.
